src/shared/codelens/localLambdaRunner.ts

```diff
@@ -310,3 +330,320 @@ export class LocalLambdaRunner {
         }
     }
 }
+
+// tslint:disable-next-line:max-line-length
+export async function run({baseBuildDir, channelLogger, codeDir, configuration, debugConfig, documentUri, handlerName, isDebug, localInvokeParams, manifestPath, runtime, samProcessInvoker, samTaskInvoker, workspaceUri}: {
+    baseBuildDir: string,
+    channelLogger: ChannelLogger,
+    codeDir: string,
+    configuration: SettingsConfiguration,
+    debugConfig: DebugConfiguration,
+    documentUri: vscode.Uri,
+    handlerName: string,
+    isDebug?: boolean,
+    localInvokeParams: LambdaLocalInvokeParams,
+    manifestPath?: string,
+    runtime: string,
+    samProcessInvoker: SamCliProcessInvoker,
+    samTaskInvoker: SamCliTaskInvoker,
+    workspaceUri: vscode.Uri,
+}): Promise<void> {
+    try {
+        // Switch over to the output channel so the user has feedback that we're getting things ready
+        channelLogger.channel.show(true)
+
+        channelLogger.info(
+            'AWS.output.sam.local.start',
+            'Preparing to run {0} locally...',
+            localInvokeParams.handlerName
+        )
+
+        const inputTemplatePath: string = await makeInputTemplate({
+            baseBuildDir,
+            codeDir,
+            documentUri,
+            handlerName,
+            runtime,
+            workspaceUri,
+        })
+        const samTemplatePath: string = await executeSamBuild({
+            baseBuildDir,
+            channelLogger,
+            codeDir,
+            inputTemplatePath,
+            manifestPath,
+            samProcessInvoker,
+
+        })
+
+        await invokeLambdaFunction({
+            baseBuildDir,
+            channelLogger,
+            configuration,
+            debugConfig,
+            samTaskInvoker,
+            samTemplatePath,
+            documentUri,
+            handlerName,
+            isDebug,
+        })
+
+    } catch (err) {
+        // TODO: logger.error?
+        console.log(err)
+        const error = err as Error
+
+        // TODO: Define standard/strategy. Sometimes err.message is/isn't part of msg "Error: {0}". Discuss.
```

Comment:
Can you elaborate on this? Is this something that we need to address before we publish?

Reply:
Updated to use `ChannelLogger` and removed comment. Comment was left from before a few discussions with you and others influenced the behavior of channel logger. Currently if errors are provided as arguments to `ChannelLogger`:

1. Output channel: will receive the results of localize substitution passed error.message
2. Logger: will receive the localized message displayed to the channel followed by the error object(s) to support logging of stack traces